cmake/compilerFlagsExiv2.cmake

```diff
@@ -1,79 +1,121 @@
+            endif()
+            if ( COMPILE_FLAG_WARRAY_BOUNDS_2 )
+                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}" -Warray-bounds=2")
+            endif()
+            if ( COMPILE_FLAG_WDUPLICATED_COND )
+                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}" -Wduplicated-cond")
+            endif()
+            if ( COMPILE_FLAG_WDUPLICATED_BRAN )
+                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}" -Wduplicated-branches")
+            endif()
+            if ( COMPILE_FLAG_WRESTRICT )
+                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}" -Wrestrict")
             endif ()
         endif ()
 
         if ( COMPILER_IS_CLANG )
+            check_c_compiler_flag(-Wassign-enum               COMPILE_FLAG_WASSIGN_ENUM       )
+            check_c_compiler_flag(-Wmicrosoft                 COMPILE_FLAG_WMICROSOFT         )
+            check_c_compiler_flag(-Wcomments                  COMPILE_FLAG_WCOMMENTS          )
+            check_c_compiler_flag(-Wconditional-uninitialized COMPILE_FLAG_WCONDITIONAL_UNINIT)
+            check_c_compiler_flag(-Wdirect-ivar-access        COMPILE_FLAG_WDIRECT_IVAR_ACCESS)
+            check_c_compiler_flag(-Weffc++                    COMPILE_FLAG_WEFFCPP            )
+            check_c_compiler_flag(-Wcomma                     COMPILE_FLAG_WCOMMA             )
+
             # https://clang.llvm.org/docs/DiagnosticsReference.html
             # These variables are at least available since clang 3.9.1
-            string(CONCAT EXTRA_COMPILE_FLAGS "-Wextra"
-                " -Wshadow"
-                " -Wassign-enum"
-                " -Wmicrosoft"
-                " -Wcomments"
-                " -Wconditional-uninitialized"
-                " -Wdirect-ivar-access"
-                " -Weffc++"
-                " -Wpointer-arith"
-                " -Wformat=2"
-                #" -Warray-bounds" # Enabled by default
-                # These two raises lot of warnings. Use them wisely
-                #" -Wconversion"
-                #" -Wold-style-cast"
-            )
-            # -Wdouble-promotion flag is not available in clang 3.4.2
-            if ( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 3.4.2 )
-                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
-                    " -Wdouble-promotion"
-                )
-            endif ()
-            # -Wcomma flag is not available in clang 3.8.1
-            if ( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 3.8.1 )
-                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
-                    " -Wcomma"
-                )
-            endif ()
-        endif ()
-    endif ()
+            if ( COMPILE_FLAG_WEXTRA )
+                string(CONCAT EXTRA_COMPILE_FLAGS " -Wextra")
+            endif()
+            if ( COMPILE_FLAG_WSHADOW )
+                string(CONCAT EXTRA_COMPILE_FLAGS " -Wshadow")
```

Comment:
I don't think this is correct. `string(CONCAT` *sets* the given variable to the concatenation of all of the following strings. So this *sets* `EXTRA_COMPILE_FLAGS` to " -Wshadow", overwriting any previous value of `EXTRA_COMPILE_FLAGS`. Since what you actually want is to *append* " -Wshadow" to any existing value of that variable, it seems simpler to me to use `set` rather than `string(CONCAT`:

```
set(EXTRA_COMPILE_FLAGS "${EXTRA_COMPILE_FLAGS} -Wshadow")
```